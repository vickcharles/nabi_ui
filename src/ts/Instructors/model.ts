export interface ExperienceType {
  id?: number;
  employer: string;
  jobTitle: string;
  fromMonth: string;
  fromYear: string;
  toMonth: string;
  toYear: string;
  location: string;
}

export enum DegreeType {
  associate = 'Associate Degree',
  bachelors = 'Bachelor\'s Degree',
  graduate = 'Graduate Degreee',
  professional = 'Professional Degree',
  certification = 'Certification',
  other = 'Other'
}

export interface EducationType {
  id?: number;
  schoolName: string;
  graduationYear: string;
  degreeType: DegreeType;
  fieldOfStudy: string;
}

export enum PlaceForLessonsOptions {
  home = 'Student\'s home',
  studio = 'Instructor\'s studio',
  online = 'Online',
}

export interface PlaceForLessonsType {
  home: boolean;
  studio: boolean;
  online: boolean;
}

export enum LessonDuration {
  thirtyMins = '30 mins',
  fortyFiveMins = '45 mins',
  sixtyMins = '60 mins',
  ninetyMins = '90 mins'
}

export interface RatesType {
  thirtyMinsRate: number;
  fortyFiveMinsRate: number;
  sixtyMinsRate: number;
  ninetyMinsRate: number;
}

export enum SkillLevel {
  beginner = 'Beginner',
  intermediate = 'Intermediate',
  advanced = 'Advanced'
}

export interface InstrumentsType {
  instrument: string;
  skillLevel: SkillLevel;
}

export interface InstructorType {
  userId: string;
  bio?: string;
  instruments?: InstrumentsType[];
  rates?: RatesType;
  placeForLessons?: PlaceForLessonsType;
  studioAddress?: string;
  education?: EducationType[];
  experience?: ExperienceType[];
}

export type IState = InstructorType[];
