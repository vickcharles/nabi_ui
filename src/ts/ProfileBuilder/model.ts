export interface InstructorState {
  userId: string;
  bio?: string;
}

export type IState = InstructorState[];
  