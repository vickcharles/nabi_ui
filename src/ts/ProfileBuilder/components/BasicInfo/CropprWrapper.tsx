import * as React from 'react';
import Cropper from 'cropperjs';
import * as logo from '../../../../assets/images/user-picture-2.png';
import  '../../../../../node_modules/cropperjs/dist/cropper.min.css';
import Button from 'material-ui/Button';
import { Theme, withStyles, WithStyles } from 'material-ui/styles';
import blur from './blur';

interface CropprState {
  isLoading: boolean;
  isCropping: boolean;
  baseImage: any;
}

interface CropprProps {
  originalImage?: any;
  imageChanged?(avatar: string): void;
}

const styles = (theme: Theme) => ({
  inIfiniteSpace: {
    display: 'block',
    top: '-100px',
    left: '-500px',
  }
});

type PropsWithStyles = CropprProps & WithStyles<'inIfiniteSpace'>;

class CropprWrapper extends React.Component<PropsWithStyles, CropprState> {
  cropperInstance: any;
  imageHolder: any;
  fileUpload: any;
  constructor(props: PropsWithStyles) {
    super(props);

    this.state = {
      isLoading: false,
      isCropping: false,
      baseImage: props.originalImage || logo,
    };

    this.cropperInstance = null;
    this.imageOnClick = this.imageOnClick.bind(this);
    this.onFileChange = this.onFileChange.bind(this);
    this.componentDidMount = this.componentDidMount.bind(this);
    this.handleCrop = this.handleCrop.bind(this);
  }

  onFileChange(e: any ) {
    let picFiles: any = e.target.files;
    let onePic: any = picFiles[0];

    if (FileReader && picFiles && picFiles.length) {
       let fr = new FileReader();
       fr.onload =  () => {
           this.setState({isCropping: true, baseImage: fr.result});
       };
       fr.readAsDataURL(onePic);
    }

  }

  componentDidUpdate() {
    if ( this.state.isCropping ) {
        // this.cropperInstance  = new Croppr(this.imageHolder, { aspectRatio : 1, startSize : [150, 150, 'px'] });
        this.cropperInstance  = new Cropper(
          this.imageHolder,
          {
            aspectRatio: 1,
            minContainerHeight: 150,
            minContainerWidth: 150,
            modal: false,
            viewMode: 1,
            minCropBoxWidth: 150,
            background: false,
          }
        );
        this.imageHolder.addEventListener('ready', ( e: any ) => {
          let crpCont = document.getElementsByClassName('cropper-container')[0];
          let crpBgd = this.imageToDataUri( this.state.baseImage, 150, 150);
          crpCont.setAttribute(
                    'style',
                    'min-width:150px;'
                    + ' min-height:150px;background-size: cover; background-image: url(\''
                    + crpBgd + '\')');
        });
    }
  }

  imageOnClick(e: any) {
    e.preventDefault();
    e.stopPropagation();
  }

  handleCrop(e: any) {
    e.preventDefault();
    e.stopPropagation();
    let newBase64 = this.cropperInstance.getCroppedCanvas(
      {
        minWidth: 150,
        maxWidth: 300,
        fillColor: '#ffffff',
        imageSmoothingQuality: 'high'
      }
    ).toDataURL();
    if (newBase64) {
      this.cropperInstance.clear();
      this.cropperInstance.destroy();
      this.cropperInstance = null;
      this.setState({isCropping: false, baseImage: newBase64});
      if ( this.props.imageChanged ) {
        this.props.imageChanged(newBase64);
      }
    }
    // Get new ImageCropped value;
  }
  componentDidMount() {
    // do Somenthing
  }

  imageToDataUri(img: string, width: number, height: number ): string {

    // create an off-screen canvas
    let canvas = document.createElement('canvas');
    let ctx: any = canvas.getContext('2d');
    let imgObj = document.createElement('img');
    imgObj.src = img;
    canvas.width = width;
    canvas.height = height;
    if ( ctx ) {
        ctx.drawImage(imgObj, 0, 0, width, height);
        let imageData = ctx.getImageData(0, 0, width, height);
        ctx.putImageData(blur(imageData, { amount: 2} ), 0, 0);
      }

    // encode image to data-uri with base64 version of compressed image
    return canvas.toDataURL();
}

  render() {
    let currentLogo = this.state.baseImage ;
    let actions = null;
    if ( this.state.isCropping ) {
      actions = (<Button color="primary" onClick={this.handleCrop} variant="raised">Change Photo</Button>);
    }
    const imgStyle = ( !this.state.isCropping ) ? {
                                                  width: '140px',
                                                  height: '140px',
                                                  maxWidth: '100%',
                                                  borderRadius: '50%',
                                                } : { maxWidth: '100%'};
    return (
      <div>
        <input
          className={this.props.classes.inIfiniteSpace}
          style={{position: 'fixed'}}
          id="uloadImageFile"
          type="file"
          onChange={this.onFileChange}
          ref={(e) => { this.fileUpload = e; }}
          disabled={this.state.isCropping}
        />
        <label style={{cursor: 'pointer', textAlign: 'center'}} htmlFor="uloadImageFile" className="inner_cropper">
             <img ref={(e) => { this.imageHolder = e; }} src={currentLogo} style={imgStyle}/>
        </label>
        <div style={{textAlign: 'right'}}>
          {actions}
        </div>
      </div>
    );
  }
}

export default withStyles(styles)<CropprProps>(CropprWrapper);
