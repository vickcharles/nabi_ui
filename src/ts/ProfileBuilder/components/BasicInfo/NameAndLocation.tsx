import * as React from 'react';
import Typography from 'material-ui/Typography';

interface NameAndLocationProps {
    firstName: string;
    zipCode: string;
  }

const NameAndLocation: React.StatelessComponent<NameAndLocationProps> = props => {
    return (
        <Typography variant="title" className="nabi-margin-top-medium nabi-margin-bottom-medium">
            {props.firstName}
            {props.zipCode}
        </Typography>
    );
};

export default NameAndLocation;
