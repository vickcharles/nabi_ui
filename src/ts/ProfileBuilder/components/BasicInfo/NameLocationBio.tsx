import * as React from 'react';
import Typography from 'material-ui/Typography';
import TextField from 'material-ui/TextField';
import Divider from 'material-ui/Divider';

interface NameLocationBioProps {
    firstName: string;
    lastName: string;
    zipCode: string;
    changeBio: (event: React.FormEvent<{}>) => void;
  }

const NameLocationBio: React.StatelessComponent<NameLocationBioProps> = props => {
  return (
    <div>
      <Typography className="nabi-text-center nabi-text-mediumbold nabi-margin-bottom-xsmall">
        {`${props.firstName} ${props.lastName}`}
      </Typography>
      
      <Typography className="nabi-text-center nabi-margin-bottom-medium">
        {props.zipCode}
      </Typography>

      <Typography className="nabi-margin-top-small" variant="body2">
        Bio *
      </Typography>

      <TextField
        id="bio"
        margin="normal"
        name="bio"
        onChange={props.changeBio}
        placeholder="What would you like your students to know  about you?"
        required={true}
        multiline={true}
        fullWidth={true}
        rows={6}
      />

      <Divider className="nabi-margin-top-small"/>
    </div>
  );
};

export default NameLocationBio;
