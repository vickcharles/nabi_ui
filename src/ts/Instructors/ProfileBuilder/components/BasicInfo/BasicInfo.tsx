import * as React from 'react';
import { Action, Dispatch } from 'redux';
import { ThunkAction } from 'redux-thunk';
import { connect } from 'react-redux';

import { UserState } from '../../../../Users/model';
import { updateUser } from '../../../../Users';
import { InstructorState } from '../../../model';
import { updateInstructor } from '../../../';
import NameLocationBio from './NameLocationBio/NameLocationBio';
import ImageUploader from './ImageUploader';

interface BasicInfoDispatchProps {
  updateInstructor: (instructor: InstructorState) => void;
  updateUser: (user: UserState) => void;
}

interface BasicInfoOwnProps { 
  user: UserState;
  changeAvatar: (id: string, avatar: string) => void;
}

interface BasicInfoState {
  bio: string;
  displayName: string;
}

interface BasicInfoProps extends
  BasicInfoDispatchProps,
  BasicInfoOwnProps { }

export class BasicInfo extends React.Component<BasicInfoProps, BasicInfoState> {
  constructor(props: BasicInfoProps) {
    super(props);

    this.state = {
      bio: '',
      displayName: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleBlurBio = this.handleBlurBio.bind(this);
    this.updateName = this.updateName.bind(this);
  }

  public componentWillMount(): void {
    this.setState({
      ...this.state,
        displayName: `${this.props.user.firstName} ${this.props.user.lastName}`
    });
  }

  public handleChange(event: any): void {
    const target = event.target;
    const value = target.value;
    const name = target.name;
    
    this.setState({
      ...this.state,
      [name]: value
    });
  }

  public handleBlurBio(event: any): void {
    const instructor: InstructorState =  {
      userId: this.props.user.id,
      bio: this.state.bio
    };
    this.props.updateInstructor(instructor);
  }

  public updateName(event: any): void {
    const user =  {
      ...this.props.user,
      displayName: this.state.displayName
    };
    this.props.updateUser(user);
  }

  public render(): JSX.Element {
    return (
      <div>
        <ImageUploader 
          imageChanged={(avatar: string) => {this.props.changeAvatar(this.props.user.id, avatar); }}
        />
        <NameLocationBio 
          user={this.props.user}
          bioBlur={this.handleBlurBio}
          bioChange={this.handleChange}
          updateName={this.updateName}
          handleChange={this.handleChange}
          displayName={this.state.displayName}
        />
      </div>
    );
  }
}

function mapDispatchToProps(
  dispatch: Dispatch<Action | ThunkAction<{}, {}, {}>>,
  _ownProps: BasicInfoOwnProps
): BasicInfoDispatchProps {
  return {
    updateInstructor: (instructor: InstructorState) => dispatch(updateInstructor(instructor)),
    updateUser: (user: UserState) => dispatch(updateUser(user))
  };
}

export default connect(null, mapDispatchToProps)(BasicInfo);
