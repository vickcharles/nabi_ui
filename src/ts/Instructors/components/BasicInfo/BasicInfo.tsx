import * as React from 'react';
import { Action, Dispatch } from 'redux';
import { ThunkAction } from 'redux-thunk';
import { connect } from 'react-redux';

import Typography from 'material-ui/Typography';

import { UserState } from '../../../Users/model';
import { updateUser } from '../../../Users';
import { InstructorState, InstrumentsType, SkillLevel } from '../../model';
import { updateInstructor } from '../../';
import NameLocationBio from './NameLocationBio/NameLocationBio';
import ImageUploader from './ImageUploader';
import Instruments from './Instruments/Instruments';
import SelectedInstrument from './Instruments/SelectedInstrument';
import Rates from './Rates';

interface BasicInfoDispatchProps {
  updateInstructor: (instructor: InstructorState) => void;
  updateUser: (user: UserState) => void;
}

interface BasicInfoOwnProps { 
  user: UserState;
  changeAvatar: (id: string, avatar: string) => void;
}

interface BasicInfoState {
  bio: string;
  displayName: string;
  instrument: string;
  skillLevel: SkillLevel;
  instruments: InstrumentsType[];
}

interface BasicInfoProps extends
  BasicInfoDispatchProps,
  BasicInfoOwnProps { }

export class BasicInfo extends React.Component<BasicInfoProps, BasicInfoState> {
  constructor(props: BasicInfoProps) {
    super(props);

    this.state = {
      bio: '',
      displayName: '',
      instrument: '',
      skillLevel: SkillLevel.beginner,
      instruments: []
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleBlurBio = this.handleBlurBio.bind(this);
    this.updateName = this.updateName.bind(this);
    this.addInstrument = this.addInstrument.bind(this);
    this.deleteInstrument = this.deleteInstrument.bind(this);
  }

  public componentWillMount(): void {
    this.setState({
      ...this.state,
        displayName: `${this.props.user.firstName} ${this.props.user.lastName}`
    });
  }

  public handleChange(event: any): void {
    const target = event.target;
    const value = target.value;
    const name = target.name;
    
    this.setState({
      ...this.state,
      [name]: value
    });
  }

  public handleBlurBio(event: any): void {
    const instructor: InstructorState =  {
      userId: this.props.user.id,
      bio: this.state.bio
    };
    this.updateInstructorCall(instructor);
  }

  public updateName(event: any): void {
    const user =  {
      ...this.props.user,
      displayName: this.state.displayName
    };
    this.props.updateUser(user);
  }

  public addInstrument(event: any): void {
    const instrumentToAdd: InstrumentsType = {
      instrument: this.state.instrument,
      skillLevel: this.state.skillLevel
    };

    if (instrumentToAdd.instrument && instrumentToAdd.skillLevel) {
      this.setState(
        { instruments: [...this.state.instruments, instrumentToAdd] },
        () => {
        const instructor: InstructorState =  {
          userId: this.props.user.id,
          instruments: this.state.instruments
        };
        this.updateInstructorCall(instructor);
      });
    }
  }

  public updateInstructorCall(instructor: InstructorState): void {
    this.props.updateInstructor(instructor);
  }

  public deleteInstrument(instrumentName: string): void {
    this.setState(
      { instruments: this.state.instruments.filter(
        instrument => instrumentName.indexOf(instrument.instrument) === -1)},
      () => {
      const instructor: InstructorState =  {
        userId: this.props.user.id,
        instruments: this.state.instruments
      };
      this.updateInstructorCall(instructor);
    });
  }

  public render(): JSX.Element {
    const selectedInstruments = this.state.instruments.map((instrument, i) => (
      <SelectedInstrument
        key={i}
        instrument={instrument.instrument}
        skillLevel={instrument.skillLevel}
        deleteInstrument={(instrumentName: string) => this.deleteInstrument(instrumentName)}
      />
    ));

    return (
      <div>
        <ImageUploader 
          imageChanged={(avatar: string) => {this.props.changeAvatar(this.props.user.id, avatar); }}
        />
        <NameLocationBio 
          user={this.props.user}
          bioBlur={this.handleBlurBio}
          bioChange={this.handleChange}
          updateName={this.updateName}
          handleChange={this.handleChange}
          displayName={this.state.displayName}
        />
        <Typography className="nabi-margin-top-small" variant="body2">
          Instruments
        </Typography>
        {selectedInstruments}
        <Instruments
          instrument={this.state.instrument}
          skillLevel={this.state.skillLevel}
          handleChange={this.handleChange}
          addInstrument={this.addInstrument}
        />
        <Rates handleChange={this.handleChange} />
      </div>
    );
  }
}

function mapDispatchToProps(
  dispatch: Dispatch<Action | ThunkAction<{}, {}, {}>>,
  _ownProps: BasicInfoOwnProps
): BasicInfoDispatchProps {
  return {
    updateInstructor: (instructor: InstructorState) => dispatch(updateInstructor(instructor)),
    updateUser: (user: UserState) => dispatch(updateUser(user))
  };
}

export default connect(null, mapDispatchToProps)(BasicInfo);
