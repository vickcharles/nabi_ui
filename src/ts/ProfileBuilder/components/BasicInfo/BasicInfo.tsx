import * as React from 'react';
import { UserState } from '../../../Registration/model';
import NameLocationBio from './NameLocationBio';
import { changeBio } from '../../actions';
import { Action, Dispatch } from 'redux';
import { ThunkAction } from 'redux-thunk';
import { connect } from 'react-redux';

interface BasicInfoStateProps {
}

interface BasicInfoDispatchProps {
  changeBio: (id: string, bio: string) => void;
}

interface BasicInfoOwnProps { 
  user: UserState;
}

interface BasicInfoProps extends
  BasicInfoStateProps,
  BasicInfoDispatchProps,
  BasicInfoOwnProps { }

export class BasicInfo extends React.Component<BasicInfoProps, {}> {
  constructor(props: BasicInfoProps) {
    super(props);

    this.handleChangeBio = this.handleChangeBio.bind(this);
  }

  public handleChangeBio(event: React.ChangeEvent<HTMLSelectElement>): void {
    this.props.changeBio(event.target.value, 'f=h');
  }

  public render(): JSX.Element {
    
    return (
    <NameLocationBio 
      firstName={this.props.user.firstName}
      lastName={this.props.user.lastName}
      zipCode={this.props.user.zipCode} 
      changeBio={this.handleChangeBio}
    />
    );
  }
}

function mapStateToProps(state: any, _ownProps: BasicInfoOwnProps): BasicInfoStateProps {
  return {
    user: state.user
  };
}

function mapDispatchToProps(
  dispatch: Dispatch<Action | ThunkAction<{}, {}, {}>>,
  _ownProps: BasicInfoOwnProps
): BasicInfoDispatchProps {
  return {
    changeBio: (id: string, bio: string) => dispatch(changeBio(id, bio))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(BasicInfo);
