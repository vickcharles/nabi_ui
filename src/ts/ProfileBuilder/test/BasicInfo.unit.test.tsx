import * as React from 'react';
import { BasicInfo } from '../components/BasicInfo/BasicInfo';
import { shallow } from 'enzyme';
import { Role, UserState } from '../../Registration/model';

describe('BasicInfo', () => {
  let wrapper: any;

  const mockUser: UserState = {
    id: '',
    firstName: 'foo',
    lastName: 'bar',
    email: 'oreo',
    password: '',
    zipCode: '',
    role: Role.instructor,
    hearAboutUs: ''
  };

  beforeEach(() => {
    wrapper = shallow(
      <BasicInfo 
        user={mockUser}
        users={[]}
      />
    );
  });

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
