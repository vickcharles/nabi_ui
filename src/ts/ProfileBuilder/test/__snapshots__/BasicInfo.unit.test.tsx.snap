// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInfo Matches snapshot 1`] = `undefined`;
